crates/store/src/state.rs

```diff
@@ -324,7 +366,14 @@ impl State {
         };
 
         // build note tree
-        let note_tree = block.body().compute_block_note_tree();
+        let note_tree_entries: Vec<_> = block
+            .body()
+            .output_notes()
+            .map(|(note_index, note)| (note_index, note.id(), *note.metadata()))
+            .collect();
+        let note_tree =
+            miden_objects::block::BlockNoteTree::with_entries(note_tree_entries.iter().copied())
+                .map_err(|e| InvalidBlockError::FailedToBuildNoteTree(e.to_string()))?;
```

Comment:
This feels like it should be a method in protocol. Or actually, that probably follows for a lot of these..